File: validation/validation-annotation-processor/src/test/java/ru/tinkoff/kora/validation/annotation/processor/testdata/Yoda.java
package ru.tinkoff.kora.validation.annotation.processor.testdata;

import org.jetbrains.annotations.Nullable;
import ru.tinkoff.kora.common.Tag;
import ru.tinkoff.kora.validation.annotation.NotEmpty;
import ru.tinkoff.kora.validation.annotation.Validated;

import java.util.List;
import java.util.Queue;

@Validated
public record Yoda(@Nullable @NotEmpty String id,
                   @NotEmpty List<Integer> codes,
                   @Validated List<Baby> babies) {}

File: validation/validation-symbol-processor/src/test/kotlin/ru/tinkoff/kora/validation/symbol/processor/testdata/Yoda.kt
package ru.tinkoff.kora.validation.symbol.processor.testdata

import org.jetbrains.annotations.Nullable
import ru.tinkoff.kora.validation.annotation.NotEmpty
import ru.tinkoff.kora.validation.annotation.Validated

@Validated
class Yoda {

    @NotEmpty
    var id: String? = null
        get() = field
        set(value) { field = value }

    @NotEmpty
    var codes: List<Int> = emptyList()
    @Validated
    var babies: List<Baby> = emptyList()
}
